node/actors/network/src/gossip/mod.rs

```diff
@@ -103,4 +129,51 @@ impl Network {
         })
         .await;
     }
+
+    /// Task that keeps hearing about new signatures and updates the L1 batch qc.
+    /// It will propagate the QC if there's enough signatures.
+    pub(crate) async fn update_batch_qc(&self, ctx: &ctx::Ctx) -> anyhow::Result<()> {
+        // FIXME This is not a good way to do this, we shouldn't be verifying the QC every time
+        // Can we get only the latest signatures?
+        loop {
+            let mut sub = self.batch_signatures.subscribe();
+            let signatures = sync::changed(ctx, &mut sub)
+                .await
+                .context("batch signatures")?
+                .clone();
+            for (_, sig) in signatures.0 {
+                self.l1_batch_qc
+                    .clone()
+                    .entry(sig.msg.number.clone())
+                    .or_insert_with(|| L1BatchQC::new(sig.msg.clone(), self.genesis()))
```

Comment:
attester might be malicious and sign invalid batches. We shouldn't construct the batch using the msg from the first signature we encounter, but rather collect the signatures for a expected batch that the node KNOWS of (node knows the expected batch because it has all the blocks of the batch).

Reply:
Okay, that sounds reasonable. I've updated it to calculate the new QC from the last one viewed by that node. This way, the node will only collect signatures for its subsequent QC. However, I'm not entirely sure if this prevents the attack from a malicious attester. In the future, should we also check hashes and batch numbers?

Reply:
Once we have the hash provided via StateKeeper interface, it should be all right (we only collect signatures for the batches that we expect). You can leave it as it is now, and we will upgrade it later. There is just a question of a race condition where the peer sends us a signature of a batch that we don't know about yet (i.e. we don't know the hash). But we will address this later as well.